app/services/update_course_stats.rb

```diff
@@ -59,7 +59,7 @@ def update_categories
   end
 
   def update_article_status
-    ArticleStatusManager.update_article_status_for_course(@course)
+    ArticleStatusManager.update_article_status_for_course(@course, update_service: self)
```

Comment:
Note this this isn't `ArticleStatusManager.new` (which corresponds to the `initialize` class method that you changed), it's a class method which also needs to be modified to accept an additional parameter. If you run the tests for the files you're working on locally, you can get quick feedback about problems like this.